### ISSUE TITLE
When a column has no data, auto collapse it

### ISSUE THREAD

**Issue by krainboltgreene** [nylas/N1#1657]
- [x] What operating system are you using? OSX

- [x] What version of N1 are you using? Version 0.4.14-53cd69b (0.4.14-53cd69b)

- [x] Does this feature exist in another mail client or tool you use? Yes.

![screen shot 2016-03-11 at 3 15 29 pm](https://cloud.githubusercontent.com/assets/334809/13718150/1d2d903c-e79c-11e5-9a78-89105898ea38.png)

I have 4 columns of data: Navigation, Collection View, Member View, Metadata View.

  - There is no meta data, so the metadata view (4th column) should auto collapse.
  - There is no member data, so the member column should collapse (3rd column).
  - There is no collection data, so the collection column should collapse (2nd column.

**Comment by bengotow** [comment]
Hey! Thanks for the feedback—I think we tried collapsing columns at one point and found it was pretty jarring because the columns would change size as soon as you clicked on an item. (For example, thread list is full-width, and you click on a thread, and the thread list jumps left and becomes narrower) I'll talk with our team and see if we can address this scenario better in a future release. If we were able to get animations in there, I think it'd be pretty nice.
